### PR TITLE
Allow regex to specify endpoints to serve on an interface

### PR DESCRIPTION
**Pull request by wintersteiger**
This adds a new configuration field to restrict endpoints to a particular set of endpoints (as paths in a regexes).

### CODE REVIEW

File: doc/host_config_schema/cchost_config.json
Comment (jumaffre) [inline doc/host_config_schema/cchost_config.json:83]: Should this be `null` when the RPC interface doesn't have ACME endorsement?

File: doc/operations/certificates.rst
Comment (jumaffre) [inline doc/operations/certificates.rst:130]:
```suggestion
    - ``challenge_server_interface``: Name of the interface that the ACME challenge server listens on. For http-01 challenges in production, this interface must be exposed publicly on port 80.
```

File: src/node/rpc/frontend.h
Comment (jumaffre) [inline src/node/rpc/frontend.h:209]: It'd be interesting to find out the impact on performance when this is set.

File: src/node/rpc/frontend.h
Comment (wintersteiger) [reply src/node/rpc/frontend.h:214]: Yeah, that was part of the plan, I didn't want a huge slow-down for existing systems/configs. Thanks for the suggestion for saving the regexes in the subsystem. The thought of saving the precompiled regexs also crossed my mind, but I wasn't sure where to put them.

**Comment by wintersteiger** [comment]
Performance: 5 runs of `ls_sgx_cft` with no regexes, with one (`[".*"]`) and with five (`["/a.*","/b.*","/c.*","/d.*","/e.*","/f.*","/.*"]`), all numbers are tx/sec:

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/cwinter/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/cwinter/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



  | None | One | Five
-- | -- | -- | --
  | 34832.04 | 31962.93 | 33976.56674
  | 34876.21 | 34791.950270870955t | 31315.60029
  | 36896.45 | 31656.6 | 31186.70236
  | 35110.67 | 34269.59 | 32726.96982
  | 33595.1 | 34909.1 | 35356.30602
  |   |   |  
Average: | 35062.09 | 33199.55 | 32912.42904



</body>

</html>

**Comment by achamayou** [comment]
@wintersteiger `std::regexp` was, and probably still is quite slow: https://stackoverflow.com/questions/70583395/why-is-stdregex-notoriously-much-slower-than-other-regular-expression-librarie

I agree that for the current restricted intended usage, this is fine.

File: doc/host_config_schema/cchost_config.json
Comment (wintersteiger) [reply doc/host_config_schema/cchost_config.json:83]: I found a way to do that in the schema! Also, full disclosure, every I time I look at a json schema, I hate them a little more.

**Comment by wintersteiger** [comment]
All done. Needs another review cycle before merging?

**Review by achamayou — APPROVED** [review]

**Pull request merged**
